tt_metal/impl/program/program.hpp

```diff
@@ -76,9 +77,9 @@ class Program {
     const std::vector< Semaphore > & semaphores() const { return semaphores_; }
 
     KernelGroup * kernels_on_core(const CoreCoord &core);
-
+    std::vector<std::shared_ptr<Buffer>> global_bufs = {};
     std::vector<KernelGroup>& get_kernel_groups();
-
+    inline void add_global_buffer(std::shared_ptr<Buffer> buf) { global_bufs.push_back(buf); }
```

Comment:
The idea of program owning buffers to extend the lifetime is great.

I think the name global is bad, if there are lifetime associated then they're not really global -- plus the program really doesn't know. 

So I suggest something like
`std::vector<std::shared_ptr<Buffer>> owned_buffer_pool` in private

And have only the methods interacting with it public:
`add_buffer(buffer)`
`release_buffers()`

Later on if needed we can make the pool a map and allow selective release like we have in Buda, if that ever becomes a use case

Reply:
APIs added